### PR TITLE
VP-1397: Imprement orders searching by created date

### PR DESCRIPTION
**Pull request by ilyawzrd**
DateSearch
ENG locales don't needed in localization file since they are incapsulated in datepicker plugin by default

### CODE REVIEW

**Comment by t13ka** [comment]
1. we should put elements in similar order that has been done in procurement:
image2020-1-9_14-53-40

**Comment by t13ka** [comment]
2. the color of the outline is picked from the theme settings or from where?
image
as you can see I have a different color on the top of the site

**Comment by t13ka** [comment]
3. can we disable the future dates in datepicker popup programmatically? I guess the ability to set these dates is useless anyway:
image

**Comment by t13ka** [comment]
4. bonus(only for checking): check please can we output the datepicker errors via our standard error popup, something like that:
image

for notification consistency

**Comment by t13ka** [comment]
5. what will happen if you use DE localization in the site? It will be the right localization changes in these date pickers?

**Comment by t13ka** [comment]
6. Could you crop the size of the datepickers and make them a little smaller e.g like the text size area?:
image

**Comment by t13ka** [comment]
7. I think without this line the page will be prettier:
image

File: client-app/src/pages/account/views/AccountOrders.vue [hunk 1]
```diff
@@ -3,6 +3,37 @@
     <loading :active.sync="isLoading" :z-index="5000"></loading>
     <account-order-details-modal :order-id="selectedOrderId">
     </account-order-details-modal>
+    <div class="row">
+      <div class="col-md-6">
+        <label for="start-datepicker">{{ $t("account.orders.from") }}</label>
+        <b-form-datepicker id="start-datepicker"
```
Comment (t13ka) [inline client-app/src/pages/account/views/AccountOrders.vue]: Start of what?
I suggest to make up a more clear name:
"orders-start-date", "orders-start-period-date", "begin-date"/"end-date" and so on..
:)

File: client-app/src/pages/account/views/AccountOrders.vue [hunk 2]
```diff
@@ -3,6 +3,37 @@
...
+                           :label-reset-button="$t('account.orders.reset')"
+                           :state="isDateValid"
+                           :locale="locale"
+                           v-bind="datepickerLabels"
+                           class="mb-2"
+                           @input="changeStartDate($event)"></b-form-datepicker>
+      </div>
+      <div class="col-md-6">
+        <label for="end-datepicker">{{ $t("account.orders.to") }}</label>
+        <b-form-datepicker id="end-datepicker"
```
Comment (t13ka) [inline client-app/src/pages/account/views/AccountOrders.vue]: the same

**Comment by t13ka** [comment]
8. I going to get all the orders on the current date. why I can't use these params?:
image
It is a server-side restriction?

File: client-app/src/pages/account/store/modules/orders-list/getters.ts [hunk 3]
```diff
@@ -16,5 +17,12 @@ export const getters: GetterTree<OrdersListState, RootState> = {
   },
   orderIsLoaded: (state: OrdersListState): boolean => {
     return state.selectedOrderIsLoaded;
+  },
+  getDatepickerLocales: (state: OrdersListState): LocaleMessages | {} => {
```
Comment (tatarincev) [inline client-app/src/pages/account/store/modules/orders-list/getters.ts]: It is the better will be set localization for date picker directly in the component controller, do not overload the state with data not related to business logic and interactions with API.

File: client-app/src/pages/account/views/AccountOrders.vue [hunk 4]
```diff
@@ -111,5 +154,28 @@ export default class AccountOrders extends Vue {
     this.selectedOrderId = orderId;
     this.$bvModal.show("orderDetailsModal");
   }
+
+  dateChanged() {
+    if (this.startDate && this.endDate) {
```
Comment (tatarincev) [inline client-app/src/pages/account/views/AccountOrders.vue]: for any sort of validation, I would like to suggest use this plugin  https://bootstrap-vue.js.org/docs/reference/validation/
you can use template-based validation rules (end must be Greater that start date ) without any code in controller level.

File: client-app/src/pages/account/views/AccountOrders.vue [hunk 5]
```diff
@@ -111,5 +154,28 @@ export default class AccountOrders extends Vue {
...
+      this.isDateValid = this.startDate < this.endDate;
+      if (this.isDateValid) {
+        this.setListConfig({ ...this.ordersList.listConfig, filters: {startDate: this.startDate, endDate: this.endDate} })
```
Comment (tatarincev) [inline client-app/src/pages/account/views/AccountOrders.vue]: You should avoid code duplication. Please try to follow the DRY principle.

File: client-app/src/pages/account/views/AccountOrders.vue (hunk 2 above)
Comment (ilyawzrd) [reply client-app/src/pages/account/views/AccountOrders.vue]: fixed in last commit

File: client-app/src/pages/account/views/AccountOrders.vue (hunk 1 above)
Comment (ilyawzrd) [reply client-app/src/pages/account/views/AccountOrders.vue]: fixed in last commit

File: client-app/src/pages/account/store/modules/orders-list/actions.ts [hunk 6]
```diff
@@ -15,6 +15,8 @@ export const actions: ActionTree<OrdersListState, RootState> = {
     searchCriteria.pageNumber = context.state.ordersList.listConfig.pageNumber;
     searchCriteria.pageSize = context.state.ordersList.listConfig.pageSize;
     searchCriteria.sort = context.state.ordersList.listConfig.filters.sort;
+    searchCriteria.startDate = context.state.ordersList.listConfig.filters.startDate;
```
Comment (tatarincev) [inline client-app/src/pages/account/store/modules/orders-list/actions.ts]: There should be a more compact form for copying certain fields from one object to another.
See example:
https://github.com/VirtoCommerce/vc-odt-mpa-theme/blob/dev/client-app/src/pages/account/views/AccountInfo.vue#L63

**Comment by asvishnyakov** [comment]
> 1. bonus(only for checking): check please can we output the datepicker errors via our standard error popup, something like that:
>    image
> 
> for notification consistency

@t13ka This popup is for unexpected unhandled errors. We usually show all validation messages in place in forms.

**Pull request merged**